src/schema.c

```diff
@@ -93,14 +91,29 @@ ldap_schema_new(TALLOC_CTX *ctx)
 LDAPObjectClass**
 ldap_schema_object_classes(const ldap_schema_t *schema)
 {
-    if (!schema)
+    if (!schema || !schema->object_classes)
     {
-        ld_error("Schema is NULL.\n");
+        ld_error("ldap_schema_object_classes - schema or object classes list is NULL!\n");
 
         return NULL;
     }
 
-    return schema->object_classes;
+    int result_size = g_hash_table_size(schema->object_classes);
+
+    LDAPObjectClass** result = talloc_array(schema, LDAPObjectClass*, result_size + 1);
```

Comment:
We need to check if allocation was a success before we start writing into that memory.